sky/setup_files/setup.py

```diff
@@ -217,7 +217,7 @@ def parse_readme(readme: str) -> str:
     # timeout of AzureCliCredential.
     'azure': [
         'azure-cli>=2.31.0', 'azure-core', 'azure-identity>=1.13.0',
-        'azure-mgmt-network'
+        'azure-mgmt-network', 'azure-storage-blob', 'msgraph-sdk'
```

Comment:
We should check these two dependencies in `clouds/azure.py::check_credentials`. Otherwise, Azure should show enabled with `sky check` but fail later when user uses Azure blob.

Something like the following:
```

        try:
            import azure.storage.blob
            import msgraph
        except ImportError as e:
            return False, (f'Failed to import Azure blob storage depdencies. Run the following commands:\n'
                           f'\n{cls._INDENT_PREFIX}  $ pip install skypilot[azure]'
                           f'{cls._INDENT_PREFIX}Details: '
                           f'{common_utils.format_exception(e)}')
```

Reply:
@Michaelvll Thanks for the catch. This is resolved at [9f48823](https://github.com/skypilot-org/skypilot/pull/3032/commits/9f488230c0aca96322d66248cc343f8a65e012f8)